front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -100,16 +106,41 @@ const ManchetteWithSpaceTimeChartWrapper = ({
 
   const [waypointsPanelIsOpen, setWaypointsPanelIsOpen] = useState(false);
 
-  const [tmpSelectedTrain, setTmpSelectedTrain] = useState(selectedTrainScheduleId);
-
-  useEffect(() => {
-    setTmpSelectedTrain(selectedTrainScheduleId);
-  }, [selectedTrainScheduleId]);
+  const projectedTrainsAndOccurrences = useMemo(
+    () =>
+      projectPathTrainResult.flatMap((train) => {
+        if (isOccurrence(train.id) && train.paced) {
+          const pacedTrainId = getPacedTrainIdFromOccurrenceId(train.id);
+          const stepInMs = Duration.parse(train.paced.step).ms;
+          const occurrencesCount = Math.ceil(Duration.parse(train.paced.duration).ms / stepInMs);
+          const occurrences = [];
+          for (let i = 0; i < occurrencesCount; i += 1) {
+            const occurrenceStartTime = dayjs(train.departureTime)
+              .add(i * stepInMs, 'ms')
+              .toDate();
+            occurrences.push({
+              ...train,
+              id: formatPacedTrainIdToOccurrenceId({
+                pacedTrainId,
+                occurrenceIndex: i,
+              }),
+              name: computeOccurrenceName(train.name, i),
+              departureTime: occurrenceStartTime,
+            });
+          }
+          return occurrences;
+        }
+        return {
+          ...train,
+        };
```

Comment:
Can you move this return at the beginning of this function ? Also you can remove the `{...}`

```const projectedTrainsAndOccurrences = useMemo(
    () =>
      projectPathTrainResult.flatMap((train) => {
        if (isTrainSchedule(train.id)) {
          return train;
        }
        // handle the paced train
        ...
    })
  );
```